Userland/Libraries/LibThreading/BackgroundAction.h

```diff
@@ -34,51 +35,71 @@ class BackgroundActionBase {
 };
 
 template<typename Result>
-class BackgroundAction final : public Core::Object
-    , private BackgroundActionBase {
-    C_OBJECT(BackgroundAction);
+struct ResultWrapper : public Core::Object {
```

Comment:
Which problem does ResultWrapper solve? It seems this is just a `Result&`, always pointing to `BackgroundAction::m_result`, only constructed in the lambda passed to `enqueue_work`, passed to the own `on_resolved` lambda which immediately unwraps it.

Reply:
You can only put Promise<Object> into the EventLoop (or do you want to have void*? probably not), therefore we need to wrap the actual result in an Object, which is what this class does.

Reply:
Ah, I see the issue, it's about `Result` not being an `Object`. Ah, this is unfortunate. Yes, I see now that ResultWrapper is necessary.